Sources/MongoSwift/ClientSession.swift

```diff
@@ -197,9 +197,22 @@ public final class ClientSession {
     /// this session has already been ended.
     internal func startIfNeeded() -> EventLoopFuture<Void> {
         switch self.state {
-        case .notStarted:
+        case let .notStarted(opTime, clusterTime):
```

Comment:
as part of making the `State` private, I moved this logic directly into here rather than having it live in `StartSessionOperation`. That operation's work is now strictly limited to calling the corresponding libmongoc method (which blocks as it checks for sessions support, hence it must be an operation.)